docs/how-to/unified_memory.rst

```diff
@@ -0,0 +1,631 @@
+.. meta::
+  :description: This chapter describes introduces Unified Memory (UM) and shows
+                how to use it in AMD HIP.
+  :keywords: AMD, ROCm, HIP, CUDA, unified memory, unified, memory, UM, APU
+
+*******************************************************************************
+Unified Memory
+*******************************************************************************
+
+Introduction
+============
+In conventional architectures, CPUs and GPUs have dedicated memory like Random
+Access Memory (RAM) and Video Random Access Memory (VRAM). This architectural
+design, while effective, can be limiting in terms of memory capacity and
+bandwidth, as continuous memory copying is required to allow the processors to
+access the appropriate data. New architectural features like Heterogeneous
+System Architectures (HSA) and Unified Memory (UM) help avoid these limitations
+and promise increased efficiency and innovation.
+
+Unified memory
+==============
+Unified Memory is a single memory address space accessible from any processor
+within a system. This setup simplifies memory management processes and enables
+applications to allocate data that can be read or written by code running on
+either CPUs or GPUs. The Unified memory model is shown in the following figure.
+
+.. figure:: ../data/unified_memory/um.svg
+
+AMD Accelerated Processing Unit (APU) is a typical example of a Unified Memory
+Architecture. On a single die, a central processing unit (CPU) is combined
+with an integrated graphics processing unit (iGPU), and both have access to a
+high-bandwidth memory (HBM) module named Unified Memory. The CPU enables
+high-performance, low-latency operations, while the GPU is optimized for high
+throughput (data processed by unit time).
+
+.. _unified memory system requirements:
+
+System Requirements
+===================
+Unified memory is supported on Linux by all modern AMD GPUs from the Vega
+series onward. Unified memory management can be achieved with managed memory
+allocation and, for the latest GPUs, with a system allocator.
+
+The table below lists the supported allocators. The allocators are described in
+the next section.
+
+.. list-table:: Supported Unified Memory Allocators
+    :widths: 40, 25, 25, 25
+    :header-rows: 1
+    :align: center
+
+    * - Architecture
+      - ``hipMallocManaged()``
+      - ``__managed__``
+      - ``malloc()``
+    * - MI200, MI300 Series
+      - ✅
+      - ✅
+      - ✅ :sup:`1`
+    * - MI100
+      - ✅
+      - ✅
+      - ❌
+    * - RDNA (Navi) Series
+      - ✅
+      - ✅
+      - ❌
+    * - GCN5 (Vega) Series
+      - ✅
+      - ✅
+      - ❌
+
+✅: **Supported**
+
+❌: **Unsupported**
+
+:sup:`1` Works only with ``XNACK=1``. First GPU access causes recoverable
```

Comment:
Add a link to XNACK documentation:
https://rocm.docs.amd.com/en/latest/conceptual/gpu-memory.html#xnack

Reply:
Nice catch!

Reply:
Added.